Spigot-API-Patches/0243-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,297 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Fruxz <cedricspitzer@outlook.de>
+Date: Fri, 18 Dec 2020 01:02:39 +0100
+Subject: [PATCH] Added SoundBuilder
+
+
+diff --git a/src/main/java/io/papermc/paper/sound/SoundBuilder.java b/src/main/java/io/papermc/paper/sound/SoundBuilder.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..afcb4cc2e68c8180a3f9b4dbdd86c70e50309bfb
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/sound/SoundBuilder.java
+@@ -0,0 +1,207 @@
++package io.papermc.paper.sound;
++
++import org.bukkit.Location;
++import org.bukkit.Sound;
++import org.bukkit.SoundCategory;
++import org.bukkit.World;
++import org.bukkit.entity.Player;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.Collection;
++import java.util.List;
++
++/**
++ * Helps prepare a sound to be sent to players.
++ */
++public interface SoundBuilder {
++
++    /**
++     * Plays the sound to all receiving players (or all). This method is safe to use
++     * Asynchronously
++     */
++    @NotNull
++    void play();
++
++    /**
++     * @return The sound going to be sent
++     */
++    @NotNull
++    Sound sound();
++
++    /**
++     * Changes what sound will be sent
++     *
++     * @param sound The sound
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder sound(@NotNull Sound sound);
++
++    /**
++     * @return The category under which the sound will be played
++     */
++    @NotNull
++    SoundCategory category();
++
++    /**
++     * Changes under which category the sound will be played
++     */
++    @NotNull
++    SoundBuilder category(@NotNull SoundCategory category);
++
++    /**
++     * @return The volume level of the sound played
++     */
++    float volume();
++
++    /**
++     * Changes the volume level of the sound played
++     *
++     * @param volume The volume
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder volume(float volume);
++
++    /**
++     * @return The pitch of the sound played
++     */
++    float pitch();
++
++    /**
++     * Changes the pitch of the sound played
++     *
++     * @param pitch The pitch
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder pitch(float pitch);
++
++    /**
++     * @return List of players who will receive the sound, or null for all in world
++     */
++    @Nullable
++    List<Player> receivers();
++
++    /**
++     * Players might be able to hear the sound even if this method returns false!
```

Comment:
Why? This is not a case of a falling tree with noone around -- if noone can hear it, noone can hear it...